index.d.ts

```diff
@@ -0,0 +1,51 @@
+export type PathTypeFn = (path: string) => Promise<boolean>;
+
+/**
+ * Checks whether the passed `path` is a file.
+ *
+ * @param path - The path to check.
+ * @returns A Promise for a boolean of whether the `path` is a file.
+ */
+export const file: PathTypeFn;
```

Comment:
These make more sense as named exports if we rename the methods with a `is` prefix. So this becomes `isFile`. Same with the other methods and in the JS code too.